### PR TITLE
Fixes for COBS without streams and older OS versions

### PR DESCRIPTION
**Pull request by spanceac**

### CODE REVIEW

File: stream.c [hunk 1]
```diff
@@ -154,7 +154,12 @@ static inline bool cobs_encode_stream_single(struct cobs_encode *encode, uint8_t
 		if (encode->u.zeros.next_zero == 0) {
 			uint8_t zero;
 			size_t num_pulled = net_buf_pull_across_fragments(&encode->buf, &zero, 1);
+#ifdef CONFIG_ASSERT
 			__ASSERT_NO_MSG(num_pulled == 1);
+#else
+			LOG_ERR("No zero found in netbuf");
+			(void)num_pulled;
+#endif
```
Comment (M1cha) [inline stream.c]: you can remove all of that and just add ARG_UNUSED(num_pulled) instead.

Comment (spanceac) [reply stream.c]: done

File: include/cobs/stream.h [hunk 2]
```diff
@@ -4,7 +4,13 @@
 #define COBS_STREAM_H_
 
 #include <stdbool.h>
+#include <version.h>
+
+#if ((KERNEL_VERSION_MAJOR) >= 3) && ((KERNEL_VERSION_MINOR) > 1)
```
Comment (M1cha) [inline include/cobs/stream.h]: this check is not correct because it'd use the wrong code for 4.0

**Review by M1cha — CHANGES_REQUESTED** [review]

File: include/cobs/stream.h (hunk 2 above)
Comment (spanceac) [reply include/cobs/stream.h]: good catch!

File: include/cobs/stream.h [hunk 3]
```diff
@@ -4,7 +4,13 @@
 #define COBS_STREAM_H_
 
 #include <stdbool.h>
+#include <version.h>
+
+#if ((KERNEL_VERSION_MAJOR) > 3) || (((KERNEL_VERSION_MAJOR) == 3) && ((KERNEL_VERSION_MINOR) > 0))
```
Comment (M1cha) [inline include/cobs/stream.h]:
```suggestion
#if (KERNEL_VERSION_MAJOR > 3) || ((KERNEL_VERSION_MAJOR == 3) && (KERNEL_VERSION_MINOR > 0))
```

Comment (M1cha) [reply include/cobs/stream.h]: but actually, I'd reverse the check. create an if that matches for legacy versions and use the latest version by default.

Comment (spanceac) [reply include/cobs/stream.h]: done

**Review by M1cha — APPROVED** [review]

**Pull request merged**